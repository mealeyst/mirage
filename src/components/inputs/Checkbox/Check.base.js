import React from 'react'
import PropTypes from 'prop-types'
import styled, { keyframes, css } from 'styled-components'

const BaseCheck  = ({ className }) => {
  return (
    <polyline className={className} points='38.75 98.75 77.75 134.75 151.25 55.25' />
  )
}

const Check = styled(BaseCheck)`
  fill: none;
  stroke-width: 20;
  stroke-linecap: round;
  stroke: ${props => props.theme.colors.white};
`

Check.propTypes = {
  theme: PropTypes.shape({
    colors: PropTypes.shape({
      white: PropTypes.string
    })
  })
}

export default Check
export { BaseCheck }
