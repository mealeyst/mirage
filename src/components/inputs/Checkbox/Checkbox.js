import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import CheckboxSVG from './CheckboxSVG.base'
import Rect from './Rect.base'
import Label from 'SRC/core/typography/Label'

class CheckboxBase extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      checked: props.checked
    }
  }

  onChange = () => {
    const { checked } = this.state
    const { input: { onChange } } = this.props
    this.setState({
      checked: !checked
    })
    onChange()
  }
  render() {
    const { className, input, children, ...props } = this.props
    const { checked } = this.state
    return (
      <Label {...props} lowercase className={className}>
        <input
          type='checkbox'
          {...input}
          onChange={this.onChange}
        />
        <CheckboxSVG {...props} checked={checked} />
        {children}
      </Label>
    )
  }
}

CheckboxBase.propTypes = {
  className: PropTypes.string,
  input: PropTypes.shape({
    value: PropTypes.bool
  }).isRequired,
  label: PropTypes.string
}

CheckboxBase.defaultProps = {
  input: {
    value: false,
    onChange: () => {}
  }
}

const Checkbox = styled(CheckboxBase)`
  display: flex;
  align-items: center;
  font-size: 1.6rem;
  font-weight: 400;

  input {
    width: 0;
    height: 0;

    opacity: 0;
  }

  input:focused + ${Rect} {
    stroke-width: 40;
  }
`

Checkbox.propTypes = {
  className: PropTypes.string,
  input: PropTypes.object.isRequired,
  label: PropTypes.string,
  theme: PropTypes.shape({
    colors: PropTypes.shape({
      rocketBlue: PropTypes.string,
      white: PropTypes.string
    })
  })
}

/** @component */
export default Checkbox
export { CheckboxBase }
