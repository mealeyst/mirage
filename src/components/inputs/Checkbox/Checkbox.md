```js
<Checkbox>
    Example Checkbox Checked
  </Checkbox>
```

```js
<Checkbox>
    Example Checkbox Unchecked
  </Checkbox>
```

```js
<Checkbox
  width='1.5rem'>
    Resized Example Checkbox
  </Checkbox>
```
