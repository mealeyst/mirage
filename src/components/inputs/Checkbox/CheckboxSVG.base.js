import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Check from './Check.base'
import Rect from './Rect.base'

const CheckboxSVG = ({ className, checked, ...props }) => {
  return (
    <svg className={className} xmlns='http://www.w3.org/2000/svg' viewBox='0 0 190 190'>
      <Rect {...props} checked={checked} />
      <Check {...props} checked={checked} />
    </svg>
  )
}

const StyledCheckboxSVG = styled(CheckboxSVG)`
  width: 2.5rem;
  height: 2.5rem;
  margin-right: .9rem;
`

StyledCheckboxSVG.propTypes = {
  checked: PropTypes.bool
}

StyledCheckboxSVG.defaultProps = {
  checked: false
}

export default StyledCheckboxSVG
export { CheckboxSVG }
