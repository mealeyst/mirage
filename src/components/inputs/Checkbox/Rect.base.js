import React from 'react'
import PropTypes from 'prop-types'
import styled, { keyframes, css } from 'styled-components'

const Rect = ({ className }) => {
  return (
    <rect className={className} x='2.5' y='2.5' width='185' height='185' rx='10' ry='10' />
  )
}

const StyledRect = styled(Rect)`
  fill: none;

  stroke: ${props => props.theme.colors.rocketBlue};
  stroke-width: 20;
  stroke-linecap: round;
`

StyledRect.propTypes = {
  theme: PropTypes.shape({
    colors: PropTypes.shape({
      rocketBlue: PropTypes.string
    })
  })
}

export default StyledRect
export { Rect }
