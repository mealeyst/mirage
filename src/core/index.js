export * from './grid'
export * from './icons'
export * from './image'
export * from './typography'
export * from './theme'
export * from './video'
