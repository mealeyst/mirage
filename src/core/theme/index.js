import React from 'react'
import { ThemeProvider } from 'styled-components'
import { Fonts } from './fontFamilies'
import theme, { BaseFontSize } from './theme'

const ROATheme = (props) => {
  return (
    <main>
      <BaseFontSize />
      <Fonts />
      <ThemeProvider theme={theme} {...props} />
    </main>
  )
}

export default ROATheme
export { theme, ROATheme }
