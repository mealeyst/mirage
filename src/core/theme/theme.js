import { injectGlobal } from 'styled-components'
import { colors } from './colors'
import media, { breakpoints, breakpointsVerbose } from './mediaQueries'
import fontFamilies from './fontFamilies'
import gridSettings from './gridSettings'

const base = injectGlobal`
  html {
    font-size: 10px;
  }
`

export default {
  breakpoints: breakpoints,
  colors: colors,
  fonts: fontFamilies,
  grid: gridSettings,
  fixedPosition: 'relative',
  media: media,
  breakpointsVerbose: breakpointsVerbose,
  base: base
}
