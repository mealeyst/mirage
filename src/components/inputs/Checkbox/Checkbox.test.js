import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { theme } from 'SRC/core/theme'
import { Checkbox, Label } from 'SRC'
import { CheckboxBase, CheckboxSVG, StyledCheckboxSVG } from './Checkbox'

const LabelText = 'Example Checkbox'

describe('(Component) Checkbox', () => {
  test('matching the snapshot', () => {
    const tree = renderer
   .create(<Checkbox theme={theme} input={{value: false}} label={LabelText}/>)
   .toJSON()
   expect(tree).toMatchSnapshot()
  })
})

describe('(Base Component) Checkbox', () => {
  test('rendering the children in the Label component', () => {
    const component = shallow(
    <CheckboxBase theme={theme} input={{value: false}} label={LabelText}/>
    )
    expect(component.find(Label).prop('children')).toContain(LabelText)
  })
  test('checking the input when input\'s value property is true', () => {
    const component = shallow(
      <CheckboxBase theme={theme} input={{value: true}} label={LabelText}/>
    )
    expect(component.find('input').prop('checked')).toBe(true)
  })
})
