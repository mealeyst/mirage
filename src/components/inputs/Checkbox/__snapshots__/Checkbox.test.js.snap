// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Checkbox matching the snapshot 1`] = `
.c1 {
  text-align: left;
  margin-bottom: 0;
  -webkit-letter-spacing: .1rem;
  -moz-letter-spacing: .1rem;
  -ms-letter-spacing: .1rem;
  letter-spacing: .1rem;
  text-transform: inherit;
  font-family: "ff-din-web",Helvetica,Arial,sans-serif;
  font-size: 1.4rem;
  font-weight: 500;
  color: #00003C;
}

.c5 {
  fill: none;
  stroke-width: 20;
  stroke-linecap: round;
  stroke: #FFFFFF;
}

.c4 {
  -webkit-animation: hBmzvO 0.25s linear forwards;
  animation: hBmzvO 0.25s linear forwards;
  stroke: transparent;
  stroke-width: 10;
  stroke-linecap: round;
  stroke-dashoffset: 200;
  fill: none;
  stroke: #0073D1;
  stroke-width: 20;
  stroke-linecap: round;
}

.c2 {
  width: 2.5rem;
  height: 2.5rem;
  margin-right: .9rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.6rem;
  font-weight: 400;
}

.c0 input {
  width: 0;
  height: 0;
  opacity: 0;
}

.c0 input:focused + .c3 {
  stroke-width: 40;
}

<label
  className="c0 c1"
  fontSize="1.4rem"
  fontWeight="500"
  letterSpacing=".1rem"
>
  <input
    checked={false}
    type="checkbox"
    value={false}
  />
  <svg
    className="c2"
    viewBox="0 0 190 190"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      className="c3 c4"
      height="185"
      rx="10"
      ry="10"
      width="185"
      x="2.5"
      y="2.5"
    />
    <polyline
      className="c5"
      points="38.75 98.75 77.75 134.75 151.25 55.25"
    />
  </svg>
  Example Checkbox
</label>
`;
